spec/support/macros.rb

```diff
@@ -0,0 +1,19 @@
+def sign_in(user = nil)
+  cookies.signed[:auth_token] = (user || Fabricate(:user)).auth_token
+end
+
+def current_user
+  User.find_by(auth_token: cookies.signed[:auth_token])
+end
+
+def sign_out
+  cookies.delete(:auth_token)
+end
+
+def feature_sign_in(user = nil)
+  user1 = user || Fabricate(:user)
```

Comment:
Why not just use the conditional assignment operator?

```
user ||= Fabricate(:user)
```

Then you can use `user` throughout the method.